jib-core/src/main/java/com/google/cloud/tools/jib/api/JavaContainerBuilder.java

```diff
@@ -245,6 +247,40 @@ public JavaContainerBuilder addDependencies(Path... dependencyFiles) throws IOEx
     return addDependencies(Arrays.asList(dependencyFiles));
   }
 
+  /**
+   * Adds the contents of a dependency JAR directory to the image. Duplicate JAR filenames are
+   * renamed with the filesize in order to avoid collisions.
+   *
+   * @param dependenciesDirectory the directory containing dependency JARs to add to the image
+   * @return this
+   * @throws IOException if adding the layer fails
+   */
+  public JavaContainerBuilder addDependencies(Path dependenciesDirectory) throws IOException {
+    if (!Files.isDirectory(dependenciesDirectory)) {
+      return addDependencies(Collections.singletonList(dependenciesDirectory));
+    }
+    return addDependencies(dependenciesDirectory, path -> true);
+  }
+
+  /**
+   * Adds the contents of a dependency JAR directory to the image. Duplicate JAR filenames are
+   * renamed with the filesize in order to avoid collisions.
+   *
+   * @param dependenciesDirectory the directory containing dependency JARs to add to the image
+   * @param pathFilter filter that determines which files (not directories) should be added
+   * @return this
+   * @throws IOException if adding the layer fails
+   */
+  public JavaContainerBuilder addDependencies(
```

Comment:
Hmm, exposing the predicate as an API method seems unnecessary.  I wonder if this shouldn't just be an internal method.